agent/agent.go

```diff
@@ -21,13 +21,41 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
```

Comment:
Are we exporting this type? FWIW, I don't think there will ever be multiple instances of agents. This is a one agent per process system.